### PR TITLE
PD-632: When onPremEnabled and in a project we're incorrectly showing project access as disabled

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

- When onPremEnabled and in a project we're incorrectly showing project access as disabled

🎟 _Jira ticket:_ [PD-632](https://jira.mongodb.org/browse/PD-632)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### CODE REVIEW

File: packages/mongo-nav/src/org-nav/OrgNav.tsx [hunk 1]
```diff
@@ -227,6 +226,14 @@ function OrgNav({
     );
   }
 
+  let displayProjectAccess: boolean;
+
+  if (onPremEnabled) {
+    displayProjectAccess = currentProjectName !== 'None' ? true : false;
+  } else {
+    displayProjectAccess = showProjectNav;
+  }
```
Comment (hswolff) [inline packages/mongo-nav/src/org-nav/OrgNav.tsx:235]: Can define the happy path as the default path.

```suggestion
  let displayProjectAccess = showProjectNav;

  if (onPremEnabled) {
    displayProjectAccess = currentProjectName !== 'None' ? true : false;
  }
```

Comment (hswolff) [reply packages/mongo-nav/src/org-nav/OrgNav.tsx:235]: Why is it being compared against the string 'None'? Someone could name their project that. We should check if the currentProjectId is set no?

Comment (brookescarlett) [reply packages/mongo-nav/src/org-nav/OrgNav.tsx:235]: Checking against none currently bc that's the value we're setting as default. You're right, smarter to check against id. Will update now.

Comment (brookescarlett) [reply packages/mongo-nav/src/org-nav/OrgNav.tsx]: Updated!

File: packages/mongo-nav/src/MongoNav.tsx [hunk 2]
```diff
@@ -237,6 +237,7 @@ function MongoNav({
           hosts={hosts}
           mode={mode}
           currentProjectName={data?.currentProject?.projectName}
+          currentProjectId={data?.currentProject?.projectId}
```
Comment (hswolff) [inline packages/mongo-nav/src/MongoNav.tsx:240]: We have `currentProjectId` on line 99 to use.

File: packages/mongo-nav/src/org-nav/OrgNav.spec.tsx [hunk 3]
```diff
@@ -340,6 +340,78 @@ describe('packages/mongo-nav/src/org-nav', () => {
     });
   });
 
+  describe('the Access Manager dropdown displays correctly', () => {
+    test('when onPrem and currentProject exist, the Project Access Manager link is not disabled and project name is displayed', () => {
```
Comment (hswolff) [inline packages/mongo-nav/src/org-nav/OrgNav.spec.tsx:344]: Maybe helps to make a describe block with onPrem is true and false to bucket those two conditionals? These sentences are looong.

File: packages/mongo-nav/src/org-nav/OrgNav.tsx [hunk 4]
```diff
@@ -227,6 +228,14 @@ function OrgNav({
     );
   }
 
+  let displayProjectAccess: boolean;
+
+  if (onPremEnabled) {
+    displayProjectAccess = !!currentProjectId;
+  } else {
+    displayProjectAccess = showProjectNav;
+  }
```
Comment (hswolff) [inline packages/mongo-nav/src/org-nav/OrgNav.tsx:237]: Is the same now?

```suggestion
  let displayProjectAccess = onPremEnabled ? !!currentProjectId : showProjectNav;
```

File: packages/mongo-nav/src/MongoNav.tsx (hunk 2 above)
Comment (brookescarlett) [reply packages/mongo-nav/src/MongoNav.tsx:240]: 👍

File: packages/mongo-nav/src/org-nav/OrgNav.spec.tsx (hunk 3 above)
Comment (brookescarlett) [reply packages/mongo-nav/src/org-nav/OrgNav.spec.tsx:344]: 👍

File: packages/mongo-nav/src/org-nav/OrgNav.tsx (hunk 4 above)
Comment (brookescarlett) [reply packages/mongo-nav/src/org-nav/OrgNav.tsx:237]: 👍

**Pull request merged**